File: .NET/Microsoft.Recognizers.Text.DateTime/Extractors/IDateTimeExtractor.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Collections.Generic;
using DateObject = System.DateTime;

namespace Microsoft.Recognizers.Text.DateTime
{
    public interface IDateTimeExtractor : IExtractor
    {
        List<ExtractResult> Extract(string input, DateObject reference);
    }
}

File: Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/extractors/IDateTimeExtractor.java
package com.microsoft.recognizers.text.datetime.extractors;

import com.microsoft.recognizers.text.ExtractResult;
import com.microsoft.recognizers.text.IExtractor;

import java.time.LocalDateTime;
import java.util.List;

public interface IDateTimeExtractor extends IExtractor {
    String getExtractorName();

    List<ExtractResult> extract(String input, LocalDateTime reference);
}
